### PR TITLE
core/bcast: don't register inactive validators

### PR DESCRIPTION
**Pull request by gsora**
Make sure to check the list of active validators before broadcasting the associated validator registration.

category: bug
ticket: none

### CODE REVIEW

File: core/bcast/recast.go [hunk 1]
```diff
@@ -101,8 +109,25 @@ func (r *Recaster) SlotTicked(ctx context.Context, slot core.Slot) error {
...
 		ctx := log.WithCtx(ctx, z.Any("duty", tuple.duty))
+
+		ethPk, err := pubkey.ToETH2()
+		if err != nil {
+			log.Error(ctx, "Can't convert pubkey to eth2 format", err)
+			continue
+		}
+
+		if _, found := activeVals[ethPk]; !found {
+			log.Debug(ctx, "Ignoring non-active validator", z.Str("pubkey", pubkey.String()))
```
Comment (corverroos) [inline core/bcast/recast.go:127]: inactive validators is expected, so this is going to be very verbose expeseically for large clusters, can remove I think.

**Review by corverroos — APPROVED** [review]

File: core/bcast/recast.go (hunk 1 above)
Comment (gsora) [reply core/bcast/recast.go:127]: yea was about to ask you for an opinion on that debug message, will remove!

File: app/app.go [hunk 2]
```diff
@@ -571,7 +571,23 @@ func wireRecaster(ctx context.Context, eth2Cl eth2wrap.Client, sched core.Schedu
...
 ) error {
-	recaster := bcast.NewRecaster()
+	recaster, err := bcast.NewRecaster(func(ctx context.Context) (map[eth2p0.BLSPubKey]struct{}, error) {
+		valList, err := eth2Cl.ActiveValidators(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		ret := make(map[eth2p0.BLSPubKey]struct{})
+
```
Comment (xenowits) [inline app/app.go:581]: nit: remove newline

**Review by xenowits — APPROVED** [review]

File: app/app.go [hunk 3]
```diff
@@ -571,7 +571,23 @@ func wireRecaster(ctx context.Context, eth2Cl eth2wrap.Client, sched core.Schedu
...
 	callback func(context.Context, core.Duty, core.PubKey, core.SignedData) error,
 ) error {
-	recaster := bcast.NewRecaster()
+	recaster, err := bcast.NewRecaster(func(ctx context.Context) (map[eth2p0.BLSPubKey]struct{}, error) {
+		valList, err := eth2Cl.ActiveValidators(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		ret := make(map[eth2p0.BLSPubKey]struct{})
```
Comment (xenowits) [inline app/app.go:580]: nit: can simply have a map[pubkey]bool
```suggestion
		ret := make(map[eth2p0.BLSPubKey]bool)
```

Comment (gsora) [reply app/app.go:580]: using `struct{}` as a dummy value for maps is a well-known practice, i don't think it needs changing

**Pull request merged**